eisbuk-admin/src/components/atoms/AttendanceCard/AttendanceCard.tsx

```diff
@@ -36,28 +56,100 @@ const AttendanceCard: React.FC<Props> = ({
   const endTime = luxonEnd.toISOTime().substring(0, 5);
 
   const timeString = `${startTime} - ${endTime}`;
+
+  // #region absentees
+
+  // type MarkAbsentee = (payload: {
+  //   slotId: Slot<"id">["id"];
+  //   userId: UserBooking["customer_id"];
+  //   isAbsent: boolean;
+  // }) => void;
+
+  const [localAbsentees, setLocalAbsentees] = useState({});
+  // const markAbsentee: MarkAbsentee = (
+  //   args: Parameters<typeof markAbsentee>[0]
+  // ) => dispatch(markAbsentee(args));
+
+  /**
+   * Handler used to toggle athlete being absent,
+   * both locally and dispatch update to firestore
+   * @param userBooking info of booking for which we're toggling attendance (user info, duration)
+   * @param isAbsent boolean, previous state of attendance
+   */
+  const toggleAbsent = (userBooking: UserBooking, isAbsent: boolean) => {
+    // update local state with absentees
+    setLocalAbsentees((state) => ({
+      ...state,
+      [userBooking.customer_id]: !isAbsent,
+    }));
+    // presist attendance data to firestore
+    markAttendance && markAttendance(userBooking.customer_id, !isAbsent);
+  };
+
+  const checkLocalChange = (
+    bookingId: UserBooking["customer_id"],
+    isAbsentInFirestore: boolean
+  ): boolean =>
+    localAbsentees[bookingId] &&
+    localAbsentees[bookingId] !== isAbsentInFirestore;
```

Comment:
All of this is unnecessary here. The local state should be kept local to `UserAttendance`. This way you don't pass too much data around, the state for `localAttendance` isn't needed anywhere outside the `UserAttendance` and in that case it would be a boolean(primitive) rather than an object(reference type), which is always better for state.
This part (`AttendanceCard`) should stay the same only it would make more sence to pass `attended` rather than `isAbsent`:
**AttendanceCard:**
```jsx
 <UserAttendance
      {...otherProps}
      attended={!absentees?.includes(user.customer_id)}
    />
```

**UserAttendance-logic:**
```typescript
// we're initializing the state with `attended` from props as they will be in sync initialy
  const [localAttended, setLocalAttended] = useState(attended);

  // no need for elaborate naming since it's the only onClick function and the code is small
  const handleClick = () => {
    const newAttended = !attended;
    setLocalAttended(newAttended);
    dispatch(markAttendance(customerId, true, bookingId)); // yes we should add booking Id to the markAttendance
  };
```
**UserAttendance-button:**
```jsx
<Button
      {...otherProps}
      colot={attended ? "secondary" : primary}
      onClick={handleClick}
    >
      {attended ? "👍" : "👎"}
    </Button>
```

Reply:
Please see how simple it is: no `hasLocalChange`, no complications and passing of `toggle` function, etc.